src/Symfony/Bundle/FrameworkBundle/Client.php

```diff
@@ -104,6 +106,41 @@ public function enableReboot()
         $this->reboot = true;
     }
 
+    /**
+     * Logins a user against a specific firewall.
+     *
+     * @param string      $username
+     * @param string      $password
+     * @param string[]    $roles
+     * @param string      $firewallName
+     * @param string|null $firewallContext
+     */
+    public function login(
+        string $username,
```

Comment:
should allow string or object actually (as the UsernamePasswordToken), otherwise this won't match the actual login process (most apps are logging the user using an object, not a string)

Reply:
Fixed